core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -1091,6 +1091,10 @@ private[spark] class TaskSetManager(
   def executorAdded() {
     recomputeLocality()
   }
+
+  def markPartitionIdAsCompletedForTaskAttempt(index: Int): Unit = {
+    successful(index) = true
```

Comment:
Should this method also make a call to `TaskSetManager.maybeFinishTaskSet()`?

Reply:
Yes, have made the necessary changes. Thank you.